### DISCUSSION TITLE
How is TransmitProfile's timer array used?

### DISCUSSION THREAD

**Discussion by theta3** [microsoft/cpp_client_telemetry#1035] [General]
# Is your question about how to consume a C++ library? No

# Is your question about specific 1DS SDK APIs or Features? Yes

This [documentation](https://github.com/microsoft/cpp_client_telemetry/blob/main/docs/custom-transmit-profile.md) states that the timer array has 3 values for `EventLatency_Normal`, `EventLatency_CostDeferred`, and `EventLatency_RealTime`. However, looking at the code https://github.com/microsoft/cpp_client_telemetry/blob/41840c389541d8ca43781fec9b5912be3467a117/lib/tpm/TransmitProfiles.cpp#L459, I only see `timers[0]` and `timers[2]` used.

Could you please confirm what's the right logic and settings here?

**Comment by lalitb** [comment]
Yes only first and third timer values are used from the rule, and it is documented here too - https://github.com/microsoft/cpp_client_telemetry/blob/main/docs/custom-transmit-profile.md#timers

> Each rule must include a timers array. The timers array should contain three values. The first value controls upload of EventLatency_Normal and EventLatency_CostDeferred events. The second value is not used and is retained for backwards compatibility. The third value controls and sets the cadence for EventLatency_RealTime events.

**Comment by theta3** [comment]
Oh that's right. I missed that middle sentence for some reason. I believe that back in the old SDK, there were also 3 timer values for `EventPriority.LOW`, `NORMAL`, and `HIGH`. When converting to this SDK, do we just use the same array values? It appears that the old SDK's second values should be moved to the first place in the new SDK?

I also see that here https://github.com/microsoft/cpp_client_telemetry/blob/41840c389541d8ca43781fec9b5912be3467a117/lib/tpm/TransmitProfiles.cpp#L377, 2nd and 1st values are different for default profile. But the documentation says 1st and 2nd values should be the same. Could you please clarify?
